client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx

```diff
@@ -67,32 +67,47 @@ function getGoogleAppsCartItems( { domains, fieldsets } ) {
 class AddEmailAddressesCard extends React.Component {
 	constructor( props ) {
 		super( props );
+
 		this.state = {
 			fieldsets: [ this.getNewFieldset() ],
 			validationErrors: null,
 		};
 	}
 
 	static getDerivedStateFromProps( props, state ) {
-		if (
-			state.fieldsets[ 0 ].firstName.value === '' &&
-			state.fieldsets.length === 1 &&
-			props.firstName !== null
-		) {
-			const { firstName, lastName } = props;
-			const fieldsets = [
-				{
-					...state.fieldsets[ 0 ],
-					firstName: { value: firstName || '' },
-					lastName: { value: lastName || '' },
-					username: { value: ( firstName && firstName.toLowerCase() ) || '' },
-				},
-			];
+		// Retrieves information from the first additional user
+		const [ { firstName: { value: firstName }, lastName: { value: lastName }, wasUserEdited } ] = state.fieldsets;
+
+		if ( wasUserEdited ) {
+			return null;
+		}
+
+		if ( props.firstName === null || props.lastName === null ) {
+			return null;
+		}
+
+		if ( props.firstName === firstName && props.lastName === lastName ) {
+			return null;
+		}
+
+		// Updates information of the first additional user with the current user settings data
+		const fieldsets = state.fieldsets.map( ( fieldset, index ) => {
+			if ( index !== 0 ) {
+				return fieldset;
+			}
+
 			return {
-				fieldsets: fieldsets,
+				...fieldset,
+				firstName: { value: props.firstName },
+				lastName: { value: props.lastName },
+				username: { value: kebabCase( props.firstName ) },
 			};
-		}
-		return null;
+		} );
+
+		return {
+			...state,
```

Comment:
I don't think you need to return the rest of the state.

Reply:
I had a look to the documentation of `getDerivedStateFromProps()` but couldn't find any mention of such feature. Could you point me to the right place? Or maybe you just meant that we don't care about `validationErrors` in this specific case?

Reply:
Weird, I'll have to test this out.  Seems odd that it would let you break the rest of state just editing one piece of it.